docs/src/configuration.md

```diff
@@ -52,26 +48,51 @@ with:
 $ mpiexecjl --project=/path/to/project -n 20 julia script.jl
 ```
 
-## Using a system-provided MPI
+## Using MPIPreferences.jl
+
+MPI.jl uses [Preferences.jl](https://github.com/JuliaPackaging/Preferences.jl) to
+allow the user to choose which MPI implementation to use for a project. This provides
+a single source of truth that can be used for `JLL` (Julia packages providing C libraries)
+that link against MPI, as well as localizing the choice of MPI implementation to a project.
+
+Users can use the provided [`use_system_binary`](@ref) or [`use_jll_binary`](@ref)
+to switch MPI implementations.
+
+### Migration from MPI.jl `v0.19`
+
+Prior to MPI.jl `v0.20` environment variables were used to configure which MPI
+library to use. These have now been removed.
 
-### Requirements
+- `JULIA_MPI_BINARY`
+- `JULIA_MPIEXEC`
+- `JULIA_MPI_INCLUDE_PATH`
+- `JULIA_MPI_CFLAGS`
+- `JULIA_MPICC`
+
+### Using a system-provided MPI
+
+#### Requirements
 
 MPI.jl requires a shared library installation of a C MPI library, supporting the MPI 3.0
 standard or later.
 
-### Building
+### Configuration
 
-To use the the system MPI, set the environment variable `JULIA_MPI_BINARY=system` and run
-`Pkg.build("MPI")`. This can be done by:
-```
-julia --project -e 'ENV["JULIA_MPI_BINARY"]="system"; using Pkg; Pkg.build("MPI"; verbose=true)'
+To use the the system MPI, run `MPI.use_system_binary()`.
+This will attempt find and identify any available MPI implementation, and create
+a file called `LocalPreferences.toml` adjacent to the current `Project.toml`.
+
+```sh
+julia --project -e 'using MPI; MPI.use_system_binary()'
 ```
-This will attempt find and identify any available MPI implementation.
+!!! note
+    You can copy `LocalPreferences.toml` to a different project, but you must list
+    `MPIPreferences` in the Project.toml either in `[deps]` or `[extras]` for the settings
+    to take effect. Due to a bug in Julia (until `v1.6.5` and `v1.7.1`), getting preferences
+    gtom transitive dependencies is broken (https://github.com/JuliaPackaging/Preferences.jl/issues/24).
+    To fix this either update your version of Julia, or add `MPIPreferences` as a direct dependency to your project.
```

Comment:
I don't understand the logic here. So do I always have to add `MPIPreferences` or only sometimes? This somehow does not align with the first part of the note.

Reply:
Yeah this is a wart... On older Julia versions the extra section didn't work properly (Need to double check this as well)